admin-api-server/src/nft/nft.service.ts

```diff
@@ -5,17 +5,37 @@ import {
   Injectable,
   Inject,
 } from '@nestjs/common';
-import { NFT_IMAGE_PREFIX, PG_CONNECTION } from 'src/constants';
+import { PG_CONNECTION } from 'src/constants';
 import { DbPool } from 'src/db.module';
 import { STMResultStatus, StateTransitionMachine, Actor } from 'roles_stm';
 import { User } from 'src/user/entities/user.entity';
 import { NftEntity } from './entities/nft.entity';
 import { RoleService } from 'src/role/role.service';
 import { S3Service } from './s3.service';
-import { NftFilterParams } from './params';
 import { Lock } from 'async-await-mutex-lock';
+import { QueryParams } from 'src/types';
+import { convertToSnakeCase, prepareFilterClause, prepareNftFilterClause } from 'src/utils';
+import { UpdateNft } from './nft.controller';
 const fs = require('fs');
 
+const getSelectStatement = (
+  whereClause = '',
+  limitClause = '',
+  sortField = 'id',
+  sortDirection = 'ASC',
+): string => `SELECT id, state, created_by, created_at, updated_at, json_agg(json_build_object('name', nftattr.name, 'value', nftattr.value)) as attributes
+FROM nft
+INNER JOIN nft_attribute nftattr on nftattr.nft_id = nft.id ${whereClause}
```

Comment:
You reverted all the queries back to what it was too with the other style of params? Is it necessary to have this query approach when using this way of params?

Asking, because if we go with this query approach, there may be SQL injection attacks present. We'd need to triple check that. And then still we may get it wrong if anything gets updated wrong down the line. 

Edit: I spent some time seeing if I can figure out a SQL injection attack, and I think it's not possible. But really the only reason why is because we're applying a single prepared statement with the `node-postgres` library. Which means that if someone at some point changes this for some reason that is probably a good reason, that then opened the whole thing up and now the database can be dropped by any malicious user. Attack looks like this: 
```
  http GET /nft filter=='{"true; delete from kanvas_user; with nft as (select 1 as id) select nft.id from nft where true": true}'`
```

So, therefore I think it is important we move away from this style of queries rather than moving back to them. Is it at all possible to keep the queries that I added, and to translate the filters and other params coming in from the REST API into input to those queries? With a set of known and accepted (whitelisted) tokens one may filter on.